### PR TITLE
Refactor `core/common.py` module

### LINKED ISSUE #10711: Refactor `core/common.py` module
We need this module to be refactored since we're using it to store a lot of commonly used variables, constants and methods in despite of where this module is called from.

What we can do here:
* Split this module into more than one.
* Move variables and constants to the related core module.

We should investigate and analyze each scenario, or propose another solution. 

Regards,
Alexis

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by Kondent**
|Related issue|
|---|
| This PR closes #10711 |

## Note
The `syscollector_endpoints` API Integration test is falling due a random error. It's currently tracked at:
* https://github.com/wazuh/wazuh/issues/12421

## Description

After we discussed this situation with the team, we decided that we need a refactor in this file but without splitting it into separated modules.

Under this development I refactored our common.py module.
What I did here:
* Changed variable names to suffice the `PEP8` formatting of constants.
* Removed unused variables.
* Fixed all involved Wazuh modules.
* Added a better organization layout for all variables, constants and methods.

## Schema
```
━ Common.py
    ┣━ Functions
    ┣━ Context variables 
    ┣━ Wazuh constants and variables
    ┃   ┣━ Commands
    ┃   ┣━ Date format
    ┃   ┣━ Extensions
    ┃   ┣━ Metadata 
    ┃   ┣━ Size and limits
    ┃   ┗━ Version
    ┗━ Wazuh path
        ┣━ Config
        ┣━ Misc
        ┣━ Sockets
        ┗━ Ruleset
```

In future developments, any variable / constant / function added to this file should use the section of the schema depending on its type. If there's no related section, a new one must be created following the current design. 

## Unit tests - Framework
<details>

<summary>Click to expand</summary>

```s
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.9.5, pytest-5.4.3, py-1.11.0, pluggy-0.13.1
rootdir: /home/kondent/git/wazuh/framework
plugins: html-2.1.1, asyncio-0.15.1, metadata-1.11.0, aiohttp-0.3.0, trio-0.7.0, cov-2.10.1
collected 1838 items                                                                                                                                                                         

wazuh/core/cluster/dapi/tests/test_dapi.py ..............................                                                                                                              [  1%]
wazuh/core/cluster/tests/test_client.py ................                                                                                                                               [  2%]
wazuh/core/cluster/tests/test_cluster.py .............................                                                                                                                 [  4%]
wazuh/core/cluster/tests/test_common.py ............................................................                                                                                   [  7%]
wazuh/core/cluster/tests/test_control.py .....                                                                                                                                         [  7%]
wazuh/core/cluster/tests/test_local_client.py .............                                                                                                                            [  8%]
wazuh/core/cluster/tests/test_local_server.py .......................                                                                                                                  [  9%]
wazuh/core/cluster/tests/test_master.py ..........................................                                                                                                     [ 11%]
wazuh/core/cluster/tests/test_server.py ...................                                                                                                                            [ 12%]
wazuh/core/cluster/tests/test_utils.py .......                                                                                                                                         [ 13%]
wazuh/core/cluster/tests/test_worker.py ...............................                                                                                                                [ 14%]
wazuh/core/tests/test_active_response.py ..............                                                                                                                                [ 15%]
wazuh/core/tests/test_agent.py ..............................................................................................................................................          [ 23%]
wazuh/core/tests/test_cdb_list.py ......................................                                                                                                               [ 25%]
wazuh/core/tests/test_common.py .......                                                                                                                                                [ 25%]
wazuh/core/tests/test_configuration.py ....................................                                                                                                            [ 27%]
wazuh/core/tests/test_database.py .............                                                                                                                                        [ 28%]
wazuh/core/tests/test_decoder.py ................                                                                                                                                      [ 29%]
wazuh/core/tests/test_exception.py ...                                                                                                                                                 [ 29%]
wazuh/core/tests/test_input_validator.py ...                                                                                                                                           [ 29%]
wazuh/core/tests/test_logtest.py ..                                                                                                                                                    [ 29%]
wazuh/core/tests/test_manager.py ...............                                                                                                                                       [ 30%]
wazuh/core/tests/test_mitre.py .............                                                                                                                                           [ 31%]
wazuh/core/tests/test_pyDaemonModule.py .....                                                                                                                                          [ 31%]
wazuh/core/tests/test_results.py ........................................                                                                                                              [ 33%]
wazuh/core/tests/test_rootcheck.py .............                                                                                                                                       [ 34%]
wazuh/core/tests/test_rule.py ......................                                                                                                                                   [ 35%]
wazuh/core/tests/test_sca.py ............                                                                                                                                              [ 36%]
wazuh/core/tests/test_security.py ............                                                                                                                                         [ 37%]
wazuh/core/tests/test_stats.py ............                                                                                                                                            [ 37%]
wazuh/core/tests/test_syscheck.py .......                                                                                                                                              [ 38%]
wazuh/core/tests/test_syscollector.py ...                                                                                                                                              [ 38%]
wazuh/core/tests/test_task.py ........                                                                                                                                                 [ 38%]
wazuh/core/tests/test_utils.py ....................................................................................................................................................... [ 46%]
.........................................................................                                                                                                              [ 50%]
wazuh/core/tests/test_vulnerability.py .                                                                                                                                               [ 50%]
wazuh/core/tests/test_wazuh_queue.py ....................                                                                                                                              [ 52%]
wazuh/core/tests/test_wazuh_socket.py ....................                                                                                                                             [ 53%]
wazuh/core/tests/test_wdb.py ........................                                                                                                                                  [ 54%]
wazuh/core/tests/test_wlogging.py ........                                                                                                                                             [ 54%]
wazuh/rbac/tests/test_auth_context.py ..                                                                                                                                               [ 54%]
wazuh/rbac/tests/test_decorators.py .........................................................................................................                                          [ 60%]
wazuh/rbac/tests/test_default_configuration.py .......................................................                                                                                 [ 63%]
wazuh/rbac/tests/test_orm.py ......................................................                                                                                                    [ 66%]
wazuh/rbac/tests/test_preprocessor.py ...........                                                                                                                                      [ 67%]
wazuh/tests/test_active_response.py ............                                                                                                                                       [ 67%]
wazuh/tests/test_agent.py .....................................................................................................................                                        [ 74%]
wazuh/tests/test_cdb_list.py .....................................................                                                                                                     [ 77%]
wazuh/tests/test_ciscat.py .................................                                                                                                                           [ 78%]
wazuh/tests/test_cluster.py .......                                                                                                                                                    [ 79%]
wazuh/tests/test_decoder.py ...................................                                                                                                                        [ 81%]
wazuh/tests/test_group.py ........                                                                                                                                                     [ 81%]
wazuh/tests/test_logtest.py ......                                                                                                                                                     [ 81%]
wazuh/tests/test_manager.py ....................................                                                                                                                       [ 83%]
wazuh/tests/test_mitre.py .......                                                                                                                                                      [ 84%]
wazuh/tests/test_rootcheck.py ..................................................                                                                                                       [ 86%]
wazuh/tests/test_rule.py ........................................................                                                                                                      [ 90%]
wazuh/tests/test_sca.py .......                                                                                                                                                        [ 90%]
wazuh/tests/test_security.py .........................................................................                                                                                 [ 94%]
wazuh/tests/test_stats.py .......                                                                                                                                                      [ 94%]
wazuh/tests/test_syscheck.py .........................                                                                                                                                 [ 96%]
wazuh/tests/test_syscollector.py ............                                                                                                                                          [ 96%]
wazuh/tests/test_task.py ............................                                                                                                                                  [ 98%]
wazuh/tests/test_vulnerability.py ...............................                                                                                                                      [100%]

====================================================================== 1838 passed, 3775 warnings in 251.62s (0:04:11) =======================================================================
```

</details>

## Unit tests - API
<details>

<summary>Click to expand</summary>

```s
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.9.5, pytest-5.4.3, py-1.11.0, pluggy-0.13.1
rootdir: /home/kondent/git/wazuh/api
plugins: html-2.1.1, asyncio-0.15.1, metadata-1.11.0, aiohttp-0.3.0, trio-0.7.0, cov-2.10.1
collected 499 items                                                                                                                                                                          

api/controllers/test/test_active_response_controller.py .                                                                                                                              [  0%]
api/controllers/test/test_agent_controller.py ..........................................                                                                                               [  8%]
api/controllers/test/test_cdb_list_controller.py ......                                                                                                                                [  9%]
api/controllers/test/test_ciscat_controller.py .                                                                                                                                       [ 10%]
api/controllers/test/test_cluster_controller.py ......................                                                                                                                 [ 14%]
api/controllers/test/test_decoder_controller.py .......                                                                                                                                [ 15%]
api/controllers/test/test_default_controller.py .                                                                                                                                      [ 16%]
api/controllers/test/test_experimental_controller.py ...............                                                                                                                   [ 19%]
api/controllers/test/test_manager_controller.py .................                                                                                                                      [ 22%]
api/controllers/test/test_mitre_controller.py .......                                                                                                                                  [ 23%]
api/controllers/test/test_overview_controller.py .                                                                                                                                     [ 24%]
api/controllers/test/test_rootcheck_controller.py ....                                                                                                                                 [ 24%]
api/controllers/test/test_rule_controller.py ........                                                                                                                                  [ 26%]
api/controllers/test/test_sca_controller.py ..                                                                                                                                         [ 26%]
api/controllers/test/test_security_controller.py ...................................................                                                                                   [ 37%]
api/controllers/test/test_syscheck_controller.py ....                                                                                                                                  [ 37%]
api/controllers/test/test_syscollector_controller.py .........                                                                                                                         [ 39%]
api/controllers/test/test_task_controller.py .                                                                                                                                         [ 39%]
api/controllers/test/test_vulnerability_controller.py ..                                                                                                                               [ 40%]
api/models/test/test_model.py ...........................                                                                                                                              [ 45%]
api/test/test_alogging.py ..........                                                                                                                                                   [ 47%]
api/test/test_authentication.py ..........                                                                                                                                             [ 49%]
api/test/test_configuration.py .........................................                                                                                                               [ 57%]
api/test/test_encoder.py ...                                                                                                                                                           [ 58%]
api/test/test_middlewares.py ............                                                                                                                                              [ 60%]
api/test/test_uri_parser.py ...                                                                                                                                                        [ 61%]
api/test/test_util.py ......................................                                                                                                                           [ 69%]
api/test/test_validator.py ..........................................................................................................................................................  [100%]

============================================================================== 499 passed, 14 warnings in 2.25s ==============================================================================
```

</details>

### CODE REVIEW

File: framework/wazuh/core/cdb_list.py [hunk 1]
```diff
@@ -273,4 +273,4 @@ def get_filenames_paths(filenames_list, root_directory=common.user_lists_path):
     list
         Full path to filenames.
     """
-    return [str(next(Path(root_directory).rglob(file), path.join(common.user_lists_path, file))) for file in filenames_list]
+    return [str(next(Path(root_directory).rglob(file), path.join(common.USER_LISTS_PATH, file))) for file in filenames_list]
```
Comment (vicferpoy) [inline framework/wazuh/core/cdb_list.py:276]: Check PEP8.

File: framework/wazuh/rule.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def get_rules(rule_ids=None, status=None, group=None, pci_dss=None, gpg13=None,
 
 
 @expose_resources(actions=['rules:read'], resources=['rule:file:{filename}'])
-def get_rules_files(status=None, relative_dirname=None, filename=None, offset=0, limit=common.database_limit, sort_by=None,
+def get_rules_files(status=None, relative_dirname=None, filename=None, offset=0, limit=common.DATABASE_LIMIT, sort_by=None,
```
Comment (vicferpoy) [inline framework/wazuh/rule.py:108]: Check PEP8.

File: framework/wazuh/core/cluster/worker.py [hunk 3]
```diff
@@ -147,7 +147,7 @@ async def sync(self, files_to_sync: Dict, files_metadata: Dict):
             # Finish the synchronization process and notify where the file corresponding to the taskID is located.
             result = await self.worker.send_request(command=self.cmd + b'_e', data=task_id + b' ' +
                                                                                    os.path.relpath(compressed_data_path,
-                                                                                                   common.wazuh_path).encode())
+                                                                                                   common.WAZUH_PATH).encode())
```
Comment (vicferpoy) [inline framework/wazuh/core/cluster/worker.py:150]: Check PEP8.

File: framework/wazuh/core/exception.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 
 from copy import deepcopy
-from wazuh.core.common import MAX_SOCKET_BUFFER_SIZE, wazuh_version as wazuh_full_version, agent_name_len_limit
+from wazuh.core.common import MAX_SOCKET_BUFFER_SIZE, WAZUH_VERSION as wazuh_full_version, AGENT_NAME_LEN_LIMIT
```
Comment (vicferpoy) [inline framework/wazuh/core/exception.py:7]: Constant variable imported as non-constant: `wazuh_full_version`

File: framework/wazuh/core/configuration.py [hunk 5]
```diff
@@ -633,20 +633,20 @@ def get_config(config_path):
...
         raise WazuhInternalError(1107)
 
     # Check if the option exists at local internal options
-    if os_path.exists(common.local_internal_options):
+    if os_path.exists(common.LOCAL_INTERNAL_OPTIONS_CONF):
         try:
-            return get_config(common.local_internal_options).get('root',
-                                                                 '{0}.{1}'.format(high_name, low_name))
+            return get_config(common.LOCAL_INTERNAL_OPTIONS_CONF).get('root',
+                                                                      '{0}.{1}'.format(high_name, low_name))
```
Comment (vicferpoy) [inline framework/wazuh/core/configuration.py:643]: Use f-strings instead. Review the whole module please.

**Review by vicferpoy — CHANGES_REQUESTED** [review]

**Review by vicferpoy — APPROVED** [review]
LGTM

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**